feat(typeahead): added clear button - all in one commit

shootermv opened the pull request.

wesleycho commented on src/typeahead/typeahead.js:
```diff
@@ -162,6 +162,7 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
 
     //pop-up element used to display matches
     var popUpEl = angular.element('<div uib-typeahead-popup></div>');
+    var clearBtnEl = angular.element('<span class="glyphicon glyphicon-remove" ng-style="{ cursor:\'pointer\',position:\'absolute\',top:positionClearBtn.top+\'px\', left:positionClearBtn.left+\'px\'}" ng-show="showClearBtn" ng-click="clearModel()"></span>');
```
It would be nicer to change this to something similar to how the popUpEl attributes are added (just the line below), i.e.
```js
var clearBtnEl = angular.element('<span></span>');
clearBtnEl.attr({
  class: 'glyphicon glyphicon-remove',
  style: 'position: absolute; cursor: pointer;',
  ngStyle: '...',
  ...
});
```

wesleycho commented:
Labeling this as a breaking change not because of an actual library breaking change, but one that changes UX.

shootermv commented:
Hi Wesley.
Changed the way of adding clearBtn element to be like the way popupElem
added.
Hope it is good
Thanks

On Sun, Aug 21, 2016 at 5:27 PM, Wesley Cho <notifications@github.com>
wrote:

> Labeling this as a breaking change not because of an actual library
> breaking change, but one that changes UX.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular-ui/bootstrap/pull/6189#issuecomment-241260663>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABRmHjPBdRzE0j_RXs-Yl37WFWLMhGKrks5qiGA9gaJpZM4JpSjI>
> .
>



-- 
http://gs500coder.blogspot.com
